Doc/library/urllib.request.rst

```diff
@@ -30,9 +30,9 @@ The :mod:`urllib.request` module defines the following functions:
    Open the URL *url*, which can be either a string or a
    :class:`Request` object.
 
-   *data* must be an object specifying additional data to be sent to the
-   server, or ``None`` if no such data is needed.  See :class:`Request`
-   for details.
+   *data* must be a :term:`bytes-like object <bytes-like object>` specifying
+   additional data to send to the server, or ``None`` if no such data is
+   needed.  See :class:`Request` for details.
```

Comment:
Files and other iterables without Content-Length should also be supported since 3.6 (https://bugs.python.org/issue12319). I think it is more helpful to leave this as a summary and include the less-useful details under the Request class.